File: zad1/java/Zad1.java
public class Zad1 {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("Invalid input");
            return;
        }
        char[] input = args[0].toCharArray();

        int sum1 = solveCaptcha(input, 1);
        int sum2 = solveCaptcha(input, input.length / 2);

        System.out.println("Part 1 : " + sum1);
        System.out.println("Part 2 : " + sum2);
    }

    private static int solveCaptcha(char[] input, int distance) {
        int sum = 0;
        for (int i = 0; i < input.length; i++) {
            char first = input[i];
            char second = input[(i + distance) % input.length];
            if (first == second) {
                sum += first - 48;
            }
        }
        return sum;
    }
}

File: zad1/kotlin/Zad1.kt
fun main(args: Array<String>) {
    val input : CharArray = args[0].toCharArray()

    val sum1 = solveCaptcha(input, 1)
    val sum2 = solveCaptcha(input, input.size / 2)
    println("Part 1: $sum1")
    println("Part 2: $sum2")
}

fun solveCaptcha(input: CharArray, distance: Int): Int {
    var sum = 0

    for (index in input.indices) {
        val first = input[index]
        val second = input[(index + distance) % input.size]
        if (first == second) {
            sum += first.toInt() - 48
        }
    }

    return sum
}